<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DialogSyntaxPatterns</class>
 <widget class="QDialog" name="DialogSyntaxPatterns">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>664</width>
    <height>742</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Syntax Highlighting Patterns</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_4">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <spacer name="horizontalSpacer_5">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QLabel" name="label_9">
       <property name="text">
        <string>Language &amp;Mode</string>
       </property>
       <property name="buddy">
        <cstring>comboLanguageMode</cstring>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="comboLanguageMode"/>
     </item>
     <item>
      <spacer name="horizontalSpacer_6">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="buttonLanguageMode">
       <property name="text">
        <string>&amp;Add/Modify Language Mode...</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Patterns</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0" rowspan="2">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QPushButton" name="buttonNew">
            <property name="text">
             <string>New</string>
            </property>
            <property name="icon">
             <iconset theme="document-new">
              <normaloff>.</normaloff>.</iconset>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="buttonDelete">
            <property name="text">
             <string>Delete</string>
            </property>
            <property name="icon">
             <iconset theme="edit-delete">
              <normaloff>.</normaloff>.</iconset>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="buttonCopy">
            <property name="text">
             <string>Copy</string>
            </property>
            <property name="icon">
             <iconset theme="edit-copy">
              <normaloff>.</normaloff>.</iconset>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="buttonUp">
            <property name="text">
             <string>Move</string>
            </property>
            <property name="icon">
             <iconset theme="go-up">
              <normaloff>.</normaloff>.</iconset>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="buttonDown">
            <property name="text">
             <string>Move</string>
            </property>
            <property name="icon">
             <iconset theme="go-down">
              <normaloff>.</normaloff>.</iconset>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QListView" name="listItems"/>
        </item>
       </layout>
      </item>
      <item row="2" column="0" colspan="2">
       <layout class="QGridLayout" name="gridLayout">
        <item row="1" column="0">
         <widget class="QComboBox" name="comboHighlightStyle"/>
        </item>
        <item row="0" column="3">
         <widget class="QLabel" name="labelPatternName">
          <property name="text">
           <string>Pattern &amp;Name</string>
          </property>
          <property name="buddy">
           <cstring>editPatternName</cstring>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="QLabel" name="labelParentPattern">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Paren&amp;t Pattern</string>
          </property>
          <property name="buddy">
           <cstring>editParentPattern</cstring>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Highlight &amp;Style</string>
          </property>
          <property name="buddy">
           <cstring>comboHighlightStyle</cstring>
          </property>
         </widget>
        </item>
        <item row="1" column="4">
         <widget class="QLineEdit" name="editParentPattern">
          <property name="enabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QLineEdit" name="editPatternName">
          <property name="clearButtonEnabled">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QPushButton" name="buttonHighlightStyle">
          <property name="text">
           <string>Add/Modify Style...</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="0" column="1">
       <widget class="QGroupBox" name="groupBox_3">
        <property name="title">
         <string>Pattern Type</string>
        </property>
        <property name="flat">
         <bool>true</bool>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QRadioButton" name="radioPass1">
           <property name="text">
            <string>Pass-&amp;1 (applied to all text when loaded or modified)</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radioPass2">
           <property name="text">
            <string>Pass-&amp;2 (parsing is deferred until text is exposed)</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radioSubPattern">
           <property name="text">
            <string>S&amp;ub-pattern (processed within start &amp; end of parent)</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radioColoring">
           <property name="text">
            <string>Colorin&amp;g for sub-expressions of parent pattern</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QGroupBox" name="groupMatching">
        <property name="title">
         <string>Matching</string>
        </property>
        <property name="flat">
         <bool>true</bool>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QRadioButton" name="radioSimpleRegex">
           <property name="text">
            <string>Highlight te&amp;xt matching regular expression</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radioRangeRegex">
           <property name="text">
            <string>H&amp;ighlight text between starting and ending REs</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="labelRegex">
     <property name="text">
      <string>&amp;Regular Expression to Match</string>
     </property>
     <property name="buddy">
      <cstring>editRegex</cstring>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QPlainTextEdit" name="editRegex">
     <property name="font">
      <font>
       <family>Monospace</family>
      </font>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="labelRegexEnd">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="text">
      <string>&amp;Ending Regular Expression</string>
     </property>
     <property name="buddy">
      <cstring>editRegexEnd</cstring>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLineEdit" name="editRegexEnd">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="font">
      <font>
       <family>Monospace</family>
      </font>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="labelRegexError">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="text">
      <string>Regular Expressi&amp;on Indicating Error in Match (Optional)</string>
     </property>
     <property name="buddy">
      <cstring>editRegexError</cstring>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLineEdit" name="editRegexError">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="font">
      <font>
       <family>Monospace</family>
      </font>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Context requirements for incremental re-parsing after changes</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>100</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLineEdit" name="editContextLines"/>
      </item>
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>&amp;lines</string>
        </property>
        <property name="buddy">
         <cstring>editContextLines</cstring>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>100</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLineEdit" name="editContextChars"/>
      </item>
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>&amp;characters</string>
        </property>
        <property name="buddy">
         <cstring>editContextChars</cstring>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>100</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QPushButton" name="buttonOK">
       <property name="text">
        <string>OK</string>
       </property>
       <property name="icon">
        <iconset theme="dialog-ok">
         <normaloff>.</normaloff>.</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="buttonApply">
       <property name="text">
        <string>Appl&amp;y</string>
       </property>
       <property name="icon">
        <iconset theme="dialog-ok-apply">
         <normaloff>.</normaloff>.</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="buttonCheck">
       <property name="text">
        <string>Chec&amp;k</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="buttonDeletePattern">
       <property name="text">
        <string>&amp;Delete</string>
       </property>
       <property name="icon">
        <iconset theme="edit-delete">
         <normaloff>.</normaloff>.</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="buttonRestore">
       <property name="text">
        <string>Restore De&amp;faults</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="buttonClose">
       <property name="text">
        <string>Close</string>
       </property>
       <property name="icon">
        <iconset theme="dialog-close">
         <normaloff>.</normaloff>.</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="buttonHelp">
       <property name="text">
        <string>&amp;Help</string>
       </property>
       <property name="icon">
        <iconset theme="help-contents">
         <normaloff>.</normaloff>.</iconset>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonClose</sender>
   <signal>clicked()</signal>
   <receiver>DialogSyntaxPatterns</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>683</x>
     <y>786</y>
    </hint>
    <hint type="destinationlabel">
     <x>642</x>
     <y>765</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
